content/blog/2017/2017-01-18-refactoring-conditional-to-pipeline.adoc

```diff
@@ -0,0 +1,475 @@
+
+
+.Pipeline
+[source, groovy]
+----
+GIT_BRANCH = sh(returnStdout: true, script: 'git rev-parse --abbrev-ref HEAD').trim()
+----
+
+==== CHANGES_SINCE_LAST_SUCCESS
+
+*Displays the changes since the last successful build.*
+
+*Parameters* (descriptions omitted):
+_reverse_, _format_, _changesFormat_, _showPaths_, _pathFormat_,
+_showDependencies_, _dateFormat_, _regex_, _replace_, _default_.
+
+Not only is the information provided by this token not exposed in Pipeline,
+the token has ten optional parameters, including format strings and regular expression
+searches. There are some examples,
+link:http://stackoverflow.com/questions/38084806/how-to-get-the-changes-since-the-last-successful-build-in-jenkins-pipeline[this answer on stackoverflow.com],
+that show how to get similar information in Pipeline,
+but nothing comes close to functional parity.
+Translating this token is beyond the scope of this post.
+
+== Slightly More Complex Example
+
+Let's do one more example that shows some of these conditions and tokens.
+This time we'll perform different build steps depending on what branch we're building.
+
+=== Freestyle
+
+Here's the configuration for Freestyle version
+(It's pretty long, click <<longer-pipeline, here>> to skip down to the Pipeline version):
+
+image::/images/post-images/2017-01-17/freestyle-conditional-long-full.png[Longer Freestyle Job, role="center", width=800]
+
+
+[[longer-pipeline]]
+=== Pipeline
+
+Here's the equivalent Pipeline:
+
+.Pipeline
+[pipeline]
+----
+// Script //
+properties ([
+    parameters ([
+        string (
+            defaultValue: '*',
+            description: '',
+            name : 'BRANCH_PATTERN'),
+        booleanParam (
+            defaultValue: false,
+            description: '',
+            name : 'FORCE_FULL_BUILD')
+    ])
+])
+
+node {
+    stage ('Build') {
+        checkout([$class: 'GitSCM',
+            branches: [[name: "origin/${BRANCH_PATTERN}"]],
+            doGenerateSubmoduleConfigurations: false,
+            extensions: [[$class: 'LocalBranch']],
+            submoduleCfg: [],
+            userRemoteConfigs: [[credentialsId: '9f926524-9fac-44df-99dd-53069380f286', url: 'https://github.com/bitwiseman/hermann']]])
+
+        GIT_BRANCH = 'origin/' + sh(returnStdout: true, script: 'git rev-parse --abbrev-ref HEAD').trim()
+        if (GIT_BRANCH == 'origin/master' || params.FORCE_FULL_BUILD) {
+
+            parallel (
+                linux: {
+                    build job: 'full-build-linux', parameters: [string(name: 'GIT_BRANCH_NAME', value: GIT_BRANCH)]
+                },
+                mac: {
+                    build job: 'full-build-mac', parameters: [string(name: 'GIT_BRANCH_NAME', value: GIT_BRANCH)]
+                },
+                windows: {
+                    build job: 'full-build-windows', parameters: [string(name: 'GIT_BRANCH_NAME', value: GIT_BRANCH)]
+                },
+                failFast: false)
```

Comment:
Minor: I guess you should be able to extract the `parameters` to a variable to reduce the duplication.